### PR TITLE
Add `dbSubnetGroup` as outputArtifact to `CreateRDSSnapshot` function

### PR DESCRIPTION
**Pull request by akankshakumari393**
## Change Overview

This PR add `dbSubnetGroup` as outputArtifact to `CreateRDSSnapshot` function

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/function/utils.go [hunk 1]
```diff
@@ -248,3 +248,27 @@ func isAuroraCluster(engine string) bool {
...
 	return false
 }
+
+func GetRDSDbSubnetGroup(ctx context.Context, rdsCli *rds.RDS, instanceID string) (*string, error) {
+	result, err := rdsCli.DescribeDBInstances(ctx, instanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Extract the dbSubnetGroup from the response
```
Comment (viveksinghggits) [inline pkg/function/utils.go:258]:
```suggestion
```

File: pkg/function/utils.go [hunk 2]
```diff
@@ -248,3 +248,27 @@ func isAuroraCluster(engine string) bool {
...
+	desc, err := rdsCli.DescribeDBClusters(ctx, instanceID)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() != rdserr.ErrCodeDBClusterNotFoundFault {
+				return nil, err
+			}
+			return nil, nil
+		}
+	}
+	// Extract the dbSubnetGroup from the response
```
Comment (viveksinghggits) [inline pkg/function/utils.go:272]:
```suggestion
```

**Review by viveksinghggits — APPROVED** [review]

File: pkg/function/create_rds_snapshot.go [hunk 3]
```diff
@@ -130,12 +132,23 @@ func createRDSSnapshot(ctx context.Context, instanceID string, dbEngine RDSDBEng
...
+	// Find dbSubnetGroup
+	var dbSubnetGroup *string
+	if !isAuroraCluster(string(dbEngine)) {
+		dbSubnetGroup, e = GetRDSDbSubnetGroup(ctx, rdsCli, instanceID)
+	} else {
+		dbSubnetGroup, e = GetRDSAuroraDbSubnetGroup(ctx, rdsCli, instanceID)
+	}
+	if e != nil {
+		return nil, errors.Wrapf(e, "Failed to get dbSubnetGroup ids. InstanceID=%s", instanceID)
+	}
```
Comment (PrasadG193) [inline pkg/function/create_rds_snapshot.go:144]: [nit] Lets use a switch case for better readability

File: pkg/function/utils.go [hunk 4]
```diff
@@ -248,3 +248,27 @@ func isAuroraCluster(engine string) bool {
...
 }
+
+func GetRDSDbSubnetGroup(ctx context.Context, rdsCli *rds.RDS, instanceID string) (*string, error) {
+	result, err := rdsCli.DescribeDBInstances(ctx, instanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Extract the dbSubnetGroup from the response
+	return result.DBInstances[0].DBSubnetGroup.DBSubnetGroupName, nil
```
Comment (PrasadG193) [inline pkg/function/utils.go:259]: For safety, we should check `if len(result.DBInstances) != 0`

File: pkg/function/utils.go [hunk 5]
```diff
@@ -248,3 +248,27 @@ func isAuroraCluster(engine string) bool {
...
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() != rdserr.ErrCodeDBClusterNotFoundFault {
+				return nil, err
+			}
+			return nil, nil
+		}
+	}
+	// Extract the dbSubnetGroup from the response
+	return desc.DBClusters[0].DBSubnetGroup, nil
```
Comment (PrasadG193) [inline pkg/function/utils.go:273]: Same here. Should verify desc.DBClusters is not empty

File: pkg/function/create_rds_snapshot.go (hunk 3 above)
Comment (akankshakumari393) [reply pkg/function/create_rds_snapshot.go:144]: Not sure if we should do that, because its would just a true false statement.

Comment (PrasadG193) [reply pkg/function/create_rds_snapshot.go:144]: It improves readability

File: pkg/function/create_rds_snapshot.go [hunk 6]
```diff
@@ -134,10 +134,11 @@ func createRDSSnapshot(ctx context.Context, instanceID string, dbEngine RDSDBEng
...
 	// Find dbSubnetGroup
 	var dbSubnetGroup *string
-	if !isAuroraCluster(string(dbEngine)) {
-		dbSubnetGroup, e = GetRDSDbSubnetGroup(ctx, rdsCli, instanceID)
-	} else {
+	switch isAuroraCluster(string(dbEngine)) {
+	case true:
 		dbSubnetGroup, e = GetRDSAuroraDbSubnetGroup(ctx, rdsCli, instanceID)
+	case false:
+		dbSubnetGroup, e = GetRDSDbSubnetGroup(ctx, rdsCli, instanceID)
```
Comment (PrasadG193) [inline pkg/function/create_rds_snapshot.go]:
```suggestion
	switch {
	case  isAuroraCluster(string(dbEngine)):
		dbSubnetGroup, e = GetRDSAuroraDbSubnetGroup(ctx, rdsCli, instanceID)
	case isRDSCluster(string(dbEngine):  // or default:
		dbSubnetGroup, e = GetRDSDbSubnetGroup(ctx, rdsCli, instanceID)
```

**Pull request merged**